### PR TITLE
Add improved joystick API

### PR DESCRIPTION
**Pull request by asarium**
This API allows to handle events from multiple joysticks but currently
only the events from the configured joystick are used.

### CODE REVIEW

**Comment by asarium** [comment]
This code is pretty old, I'll need to clean it up before it can be merged.

**Comment by asarium** [comment]
Ok, code has been cleaned up a bit. Now this is ready to be merged.

**Comment by asarium** [comment]
Branch has been rebased onto master.

File: code/sound/voicerec.cpp [hunk 1]
```diff
@@ -41,6 +41,10 @@
 #include "osapi/osapi.h"
 #include "playerman/player.h"
 #include "ship/ship.h"
+#include "cfile/cfile.h"
+#include "osapi/osapi.h"
+
+#include <SDL_syswm.h>
```
Comment (z64555) [inline code/sound/voicerec.cpp]: Duplicate include of <SDL_syswm.h>

Also, these changes here (code/sound/voicerec.cpp:44) don't appear to be relevant to the PR.

File: code/osapi/osapi.cpp [hunk 2]
```diff
@@ -103,11 +103,6 @@ namespace
 	}
 }
 
-
-// ----------------------------------------------------------------------------------------------------
```
Comment (z64555) [inline code/osapi/osapi.cpp]: Minor irrelevant omission of comment block.

**Comment by asarium** [comment]
I have resolved the issues you mentioned.
